sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -1081,12 +1081,8 @@ private[lf] object SBuiltinFun {
             // (although the authority might be being restricted)
             doExe(Some(authorizers))
           } else {
-            // additional authority *is* required; ask the ledger
-            machine.needAuthority(
-              NameOf.qualifiedNameOfCurrentFunc,
-              holding = holding,
-              requesting = requesting,
-              continue = () => doExe(Some(authorizers)),
+            Control.Error(
+              IE.RejectedAuthorityRequest(holding = holding, requesting = requesting)
```

Comment:
So we're still allowing authority to be reduced, which is why this error still exists? Verses just using `FailedAuthorization`

Reply:
> So we're still allowing authority to be reduced, which is why this error still exists?

Reply:
Finally I simplify the exception. 
I just used FailedAuthorization, i.e.  we use this exception when the `controllers union  choiceAuthorizer` are not a subset of the `authorized parties`.

With this change I could simplify a bit the BeginExercise.